SCons/Variables/PackageVariable.py

```diff
@@ -86,7 +86,7 @@ def _converter(val: str | bool, default: str) -> str | bool:
     if lval in ENABLE_STRINGS:
         # Can't return the default if it is one of the enable/disable strings;
         # test code expects a bool.
-        if default in ENABLE_STRINGS:
+        if default in ENABLE_STRINGS + DISABLE_STRINGS:
```

Comment:
This looks right to me.

Reply:
Seems wrong.. if the default is a DISABLE_STRING shouldn't it return false?

Reply:
This code seems more correct? https://bugs.gentoo.org/950584#c15

Reply:
> Seems wrong.. if the default is a DISABLE_STRING shouldn't it return false?

no, look up to line 86 - we've got a 'true' / enable string (from probably the command line). We want to enable not disable at this point.

Reply:
> This code seems more correct? https://bugs.gentoo.org/950584#c15

It's not, actually.  You already know you *must* return a truthy value: the variable's value is one of the "enabling" strings, so you can't return `False`.  The inner check for the default being an enabling/disabling string is to make sure we can tell if the default is a path string or not.

Someone thought it was a cool idea to define an interface where if you set a variable to something, it might actually produce something different. Fun.

Reply:
> Someone thought it was a cool idea to define an interface where if you set a variable to something, it might actually produce something different. Fun.

By which I mean: 
```py
SConstruct:
...
pkg = PackageVariable(name='FOO', ... default='/path/to/foo')
...

$ scons FOO=yes   # final value will be /path/to/foo, not True
```

Reply:
Gotcha.